youtube_dl/extractor/youporn.py

```diff
@@ -153,6 +153,8 @@ def _real_extract(self, url):
         thumbnail = self._search_regex(
             r'(?:imageurl\s*=|poster\s*:)\s*(["\'])(?P<thumbnail>.+?)\1',
             webpage, 'thumbnail', fatal=False, group='thumbnail')
+        duration = int_or_none(self._html_search_meta(
+            'video:duration', webpage, 'duration', default=False, fatal=False))
```

Comment:
If you have `default`, `fatal` is not used.

Reply:
I did not know, thanks for the tip! I learn something new every time I do something with youtube-dl :) I've removed the `default`, following the line of the rest of the code in this file.

Reply:
I know what you mean :) I think `default=None` is actually better (without `fatal=False`). `fatal=False` makes it emit a warning and return `None`, while `default=None` will just silently return `None`. For optional metadata like `duration`, I think it's better not to have a warning.